### PR TITLE
Update Form.tsx and make submit method public

### PR DESCRIPTION
**Pull request by popmanhe**
### Reasons for making this change

The submit method is not a public method. Somehow, it does not work in some senarios.

If this is related to existing tickets, include links to them as well. Use the syntax `fixes #[issue number]` (ex: `fixes #123`).

If your PR is non-trivial and you'd like to schedule a synchronous review, please add it to the weekly meeting agenda: https://docs.google.com/document/d/12PjTvv21k6LIky6bNQVnsplMLLnmEuypTLQF8a-8Wss/edit

### Checklist

- [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x ] **I'm adding or updating code**
  - [x ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by heath-freenome** [comment]
@popmanhe Can you please update your PR after running `npm run cs-format`? The linter is failing

**Comment by popmanhe** [comment]
npm run cs-format  executed

**Comment by heath-freenome** [comment]
@popmanhe And can you also update the `CHANGELOG.md` for `core` explaining the fix and referencing the issue. Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ should change the heading of the (upcoming) version to include a major version b
 ## @rjsf/core
 
 - Removed `dateElementProps` function implementation, and replaced it with `getDateElementProps` from `@rjsf/utils`.
+- Modify submit methond and make it a public method.
```
Comment (heath-freenome) [inline CHANGELOG.md:27]:
```suggestion
- Modify submit method and make it a public method, fixing [#4015](https://github.com/rjsf-team/react-jsonschema-form/issues/4015)
```

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**